### PR TITLE
WIP: Adds QPS ticket request

### PR DESCRIPTION
**Pull request by schmetzyannick**

### CODE REVIEW

File: src/utils/ConfigUtil.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as _ from "lodash";
 import * as dotenv from "dotenv";
+import fileExists from "file-exists";
```
Comment (schmetzyannick) [inline src/utils/ConfigUtil.ts:3]: fs has exsitsSync

File: src/utils/ConfigUtil.ts [hunk 2]
```diff
@@ -24,13 +26,47 @@ export class ConfigUtil {
...
+    public static getQpsUri(): string {
+        dotenv.config({path: resolve(process.cwd(), ".env")});
+        return _.get(process.env, "QPS_URI", "https://qlikserver.example.org:4243/qps/customVirtualProxyPrefix");
+    }
+
+    /**
+     * Returns the configured certificate path for authentication.
+     */
+    public static getClientPfx(): Buffer {
+        dotenv.config({path: resolve(process.cwd(), ".env")});
```
Comment (schmetzyannick) [inline src/utils/ConfigUtil.ts:48]: Can we just do this once on startup?

File: src/utils/QpsUtil.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        const ticketUrl = urljoin(ConfigUtil.getQpsUri(), "ticket?xrfkey=" + xrfKey);
+        const data = JSON.stringify({UserId: user, UserDirectory: directory, Attributes: []});
+        const response = await axios({
+            url: ticketUrl,
+            method: "POST",
+            headers: {
+                "X-Qlik-Xrfkey": xrfKey,
+                "Content-Type": "application/json",
+            },
+            httpsAgent: new https.Agent({
```
Comment (schmetzyannick) [inline src/utils/QpsUtil.ts:27]: is there any chance that the qps in running under http?

File: src/utils/QpsUtil.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                passphrase: ConfigUtil.getClientPfxPassword(),
+            }),
+            data,
+        });
+        return _.get(response, "data.Ticket", "");
+    }
+
+    /**
+     * Generates a Xrfkey which must be used prevent cross-site attacks.
+     * @returns {string} The Xrfkey.
```
Comment (schmetzyannick) [inline src/utils/QpsUtil.ts:39]: No types in comments

File: src/utils/QpsUtil.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    private static generateXrfKey(): string {
+        /* eslint-enable max-len*/
```
Comment (schmetzyannick) [inline src/utils/QpsUtil.ts:42]: .

File: src/utils/ConfigUtil.ts (hunk 2 above)
Comment (hansnull) [reply src/utils/ConfigUtil.ts:47]: Cache implemented

**Pull request merged**